rule.py

```diff
@@ -1,100 +1,254 @@
-    
-    def elephant(self, old_pos, new_pos):
-        pass 
+    def is_interfered(self, old_pos, new_pos):
+        the_one = ['horse', 'elephant']
+        if self.get_piece(old_pos).type in the_one:  
+            if self.get_piece(old_pos).type == the_one[0]: 
+                return self.board_map_2d[new_pos.y][new_pos.x] != self._IS_EMPTY and self.is_interfered_horse(old_pos, new_pos)
+            elif self.get_piece(old_pos).type == the_one[1]:
+                return self.board_map_2d[new_pos.y][new_pos.x] != self._IS_EMPTY and self.is_interfered_elephant(old_pos, new_pos)
+            else:
+                raise ValueError('MovRuleSpecial_5: Unknown piece. Piece is not either horse or elephant.')
+        else:
+            return self.board_map_2d[new_pos.y][new_pos.x] != self._IS_EMPTY
 
-    def advisor(self, old_pos, new_pos):
-        pass 
+    def is_interfered_horse(self, old_pos, new_pos):
+        buffer_x = new_pos.x - old_pos.x
+        buffer_y = new_pos.y - old_pos.y
+        ki_da    = Pos()
+        if buffer_y == 2: 
+            ki_da.y = old_pos.y + buffer_y - 1
+        elif buffer_y == -2: 
+            ki_da.y = old_pos.y + buffer_y + 1
+        elif abs(buffer_y) == 1: 
+            ki_da.y = old_pos.x + buffer_y
+        else: 
+            return False 
 
-    def general(self, old_pos, new_pos):
-        pass 
+        if buffer_x == 2: 
+            ki_da.x = old_pos.x + buffer_x - 1
+        elif buffer_x == -2:
+            ki_da.x = old_pos.x + buffer_x + 1    
+        elif abs(buffer_x) == 1: 
+            ki_da.x = old_pos.x + buffer_x 
+        else: 
+            return False 
+        
+        return self.board_map_2d[ki_da.y][ki_da.x] == self._IS_EMPTY 
+
+    def is_interfered_elephant(self, old_pos, new_pos):
+        return self.board_map_2d[old_pos.y + int((new_pos.y - old_pos.y)/2)][old_pos.x + int((new_pos.x - old_pos.x)/2)]
+
+    def is_enemy(self, Piece_1, Piece_2):
+        return Piece_1.team != Piece_2.team
+
+    def get_piece(self, pos):
+        return self.board_map_2d[pos.y][pos.x]
+
+    def get_eaten(self, eater, old_pos, new_pos): 
+        self.board_map_2d[new_pos.y][new_pos.x] = eater 
+        self.board_map_2d[old_pos.y][old_pos.x] = self._IS_EMPTY
+
+    def move_allowed(self, old_pos, new_pos): 
+        '''
+        Call functions from moving_ruleset by the names.
+
+        Soldier case is not done. 
+        TO DO: define Piece object type further into smaller categories of soldiers 
+        '''
+        
+        if is_in_board(new_pos): 
+            if not is_interfered(board_map, old_pos, new_pos):
+                if self.get_piece(old_pos).type != 'soldier':
+                    return  getattr(moving_ruleset, self.get_piece(old_pos).type)(old_pos, new_pos) and self.you_shall_not_pass(old_pos, new_pos)
+                elif self.get_piece(old_pos).pass_river: 
+                    if self.is_in_northern_palace(self.get_piece(old_pos).his_pos[0]):
```

Comment:
Why is the soldier be inside northern palace? Maybe in the north side?. Also, if you want to get current position of the piece, it is better to use the method get_current_pos() then to use his_pos[0] directly. His_pos is mostly used to access all of the previous move.

Reply:
The same for all other soldier case? Should it be in northern side instead?

Reply:
yeah, weird day leads weird mistakes :D